deployment/helm/skaha/image-cache/cache-images.sh

```diff
@@ -0,0 +1,62 @@
+harborHosts=$(echo $HARBOR_HOST | tr ' ' '\n')
+
+URL="https://$harborHost/api/v2.0/projects?page_size=100"
+PROJECT_URL="https:/$harborHost/api/v2.0/projects"
+
+# checking if redis instance available or not
+if ! redis-cli -h $REDIS_HOST -p $REDIS_PORT ping > /dev/null 2>&1; then
+    echo "Redis server is not running. Please start the Redis server and try again."
+    exit 1
+fi
+
+# Fetch the data from the given URL
+redis-cli -h $REDIS_HOST -p $REDIS_PORT del temp
+
+echo "$harborHosts" | while read -r harborHost; do
+    URL="https://$harborHost/api/v2.0/projects?page_size=100"
+    PROJECT_URL="https://$harborHost/api/v2.0/projects"
+
+    echo "fetching the images host: $URL"
+    response=$(curl -k $URL)
+
+    # Parse the response and iterate over the list
+    echo $response | jq -c '.[]' | while read -r project; do
+        project_name=$(echo $project | jq -r '.name')
+
+        project_data=$(curl -k -s "$PROJECT_URL/$project_name/repositories?page_size=-1")
+
+        # Null check for project_data
+        if [ -z "$project_data" ] || [ "$project_data" == "null" ]; then
+            continue
+        fi
+
+        echo $project_data | jq -c '.[]' | while read -r repo; do
+            repo_name=$(echo $repo | jq -r '.name')
+
+            name=$(echo $repo_name | awk -F'/' '{print $NF}')
+            repo_data=$(curl -k -s "$PROJECT_URL/$project_name/repositories/$name/artifacts?detail=true&with_label=true&page_size=-1")
+
+            echo $repo_data | jq -c '.[]' | while read -r artifact; do
+                tag=$(echo $artifact | jq -r .tags[0].name)
+
+                if [ -z "$tag" ] || [ "$tag" == "null" ]; then
+                    continue
+                fi
+
+                image_id="$harborHost/$name:$tag"
```

Comment:
We display the Project Name with our images, so this should read:
`$harborHost/$project_name/$name:$tag`

Reply:
Done